xbmc/filesystem/ISO9660File.cpp

```diff
@@ -23,18 +23,19 @@ bool CISO9660File::Open(const CURL& url)
   if (m_iso && m_stat)
     return true;
 
-  if (!m_iso->open(url.GetHostName().c_str()))
+  if (!m_iso->open_fuzzy(url.GetHostName().c_str()))
     return false;
 
   m_stat.reset(m_iso->stat(url.GetFileName().c_str()));
 
   if (!m_stat)
     return false;
 
-  if (!m_stat->p_stat)
-    return false;
+  if (m_stat->p_stat != nullptr)
```

Comment:
You don't need to be explicit about the `!= nullptr` here as the pointer check is perfectly valid without it.

I would be careful with this change as `m_start` may be used uninitialized. I'm not even sure if it's valid to start reading at `m_start = 0` (it may be valid I just don't know).

I would think p_stat needs to be valid as it's needed to check the length of the file, the size, and the type.

Reply:
Comparing to nullptr is a personal preference. @garbear usually requests it and I've started using it ever since. I can change it back if needed.

For the second part, I'm a bit lost so your help is greatly appreciated. What I can tell is that this worked fine for all my dvds and iso's. For the actual physical discs, pStat was always null.

Reply:
it's personal preference really but completely valid in c++. see, https://stackoverflow.com/a/17772117

I would think if `m_stat->p_stat` is nullptr there would be more problems:
https://github.com/xbmc/xbmc/blob/master/xbmc/filesystem/ISO9660File.cpp#L55
https://github.com/xbmc/xbmc/blob/master/xbmc/filesystem/ISO9660File.cpp#L57
https://github.com/xbmc/xbmc/blob/master/xbmc/filesystem/ISO9660File.cpp#L108

I would make sure that `CISO9660File` is actually being used in other cases as in my experience libdvd is used instead to play the file.

`CISO9660File` is only needed to access an iso9660 file that doesn't have dvd like contents.

Reply:
Yeap you're right, the iso9660 is not used further to read the file. The problem happens here:

https://github.com/xbmc/xbmc/blob/53fd5ef8b64e4d2d6494c11938a5b68026700523/xbmc/storage/MediaManager.cpp#L721

This calls `CFile::Exists` that, due to the `iso9660://` special path ends up calling `CISO9669File::Exists` and all that method does is call Open(): https://github.com/xbmc/xbmc/blob/2d993a1e09427653dec6f2a7ab8972292a3e49e2/xbmc/filesystem/ISO9660File.cpp#L118

Maybe the proper fix for this is to change the implementation of Exists, unsure it is used to initialize anything though.